vignettes/analyzing-load-test-logs.Rmd

```diff
@@ -12,31 +12,268 @@ vignette: >
 
+
+# Website Anatomy
+
+Websites are comprised of two main parts:
+
+* Main [HTML](https://en.wikipedia.org/wiki/HTML) document
+* Supporting files, such as images, [CSS](https://en.wikipedia.org/wiki/Cascading_Style_Sheets) files (webpage style), and [JavaScript](https://en.wikipedia.org/wiki/JavaScript) files (page interactivity).
+
+The main HTML document must be processed to determine which supporting files to retrieve.  This is why you will always see the first event as `Requesting the Homepage`.  Many (~ 10) of the following events will be to request specific CSS or JavaScript files.  After loading the CSS files, the browser will then style the webpage.
+
+Once the JavaScript files have finished loading, communication to a data server can begin.  For Shiny applications, this involves communicating over a [WebSocket](https://en.wikipedia.org/wiki/WebSocket).  A WebSocket is a single, constant line of communication between the Shiny server and the Shiny application in the browser.  The WebSocket will notify the Shiny server about which input values have changed by a user and will also notify the Shiny application about which outputs need to been updated.  This communication process will continue until the WebSocket is closed, typically the last line of the `recording.log` file.
+
+As an example, the basic [`01_hello` gallery shiny app](https://gallery.shinyapps.io/001-hello/) loads the homepage, loads more than 20 supporting CSS and JavaScript files, and then begins WebSocket communication with the supporting Shiny server.
+
+
+# Report Output
+
+The data provided by `shinyloadtest::tidy_loadtest` can be used to directly create a stand alone HTML report.  This report contains six major sections of diagnostic information.
+
+```{r, include = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  `1 user` = "../test_sessions/demo1",
+  `4 users` = "../test_sessions/demo4",
+  `16 users` = "../test_sessions/demo16"
+)
+# ignore <- shinyloadtest::make_report(df, "report.html", verbose = FALSE, open_browser = FALSE)
+```
+```{r, eval = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  "1 user" = "./demo1",
+  "4 users" = "./demo4",
+  "16 users" = "./demo16"
+)
+shinyloadtest::make_report(df, "report.html")
+```
+
+## Sessions
+
+The `Sessions` tab of the shinyloadtest report displays the each simulated session performed by each user.  Each run displays their respective [gantt chart](https://en.wikipedia.org/wiki/Gantt_chart) by clicking on the named buttons near the top of the webpage.
+
+```{r, echo = FALSE, fig.height = 3}
+suppressPackageStartupMessages(library(dplyr, quiet = TRUE))
+suppressPackageStartupMessages(library(shinyloadtest, quiet = TRUE))
+df %>%
+  filter(run == "4 users") %>%
+  plot_gantt()
+```
+
+Sessions that are desaturated in color are sessions that have started before (warmup) or ended after (cooldown) the maintenance period.  The maintenance period is marked with vertical dotted lines spanning every simulated user.
```

Comment:
I'd recommend the following rewording which clarifies and defines "maintenance" period:

`**Maintenance period**: During a load test, user sessions are not all started at exactly the same time. Instead, user requests are staggered over a warmup period. Once the targetted number of concurrent sessions is reached, the maintenance period begins during which time shinycannon maintains the number of user sessions. After the desired test duration is reached, user sessions begin ending during a cooldown period. Sessions that are desaturated in color occurred during the warmup or cooldown. The maintenance period is marked with vertical dotted lines spanning every simulated user.